src/decision-engine/index.js

```diff
@@ -45,6 +47,7 @@ class DecisionEngine {
       outstanding--
       batch.push(b)
       size += b.data.byteLength
+      total -= b.data.byteLength
 
       if (size >= MAX_MESSAGE_SIZE ||
```

Comment:
Is there a bug here? We add to the batch until size is greater than (or equal to) `MAX_MESSAGE_SIZE` and then we send that batch...surely that message is then too big to send?

Reply:
That's a good question - `MAX_MESSAGE_SIZE` is set to 256k, but in reality we should be able to send messages an order of magnitude larger than that. So I think probably this is by design: we want to send _at least_ MAX_MESSAGE_SIZE and a little bit more. This is actually how go-bitswap does it.

I'm currently working on a PR to change how all this works so my inclination is to ignore for this PR and fix in my next PR, does that sound reasonable?

Reply:
👍 SGTM